tests/utils/test_ellipsoid_utils.py

```diff
@@ -2,17 +2,203 @@
+        camera2 = PinholeCameraCal3Bundler(Pose3(Rot3(np.eye(3)), np.array([1, -1, 2])), default_intrinsics)
+
+        sample_data.add_camera(0, camera0)
+        sample_data.add_camera(1, camera1)
+        sample_data.add_camera(2, camera2)
+
+        # Add 6 tracks to sample_data
+
+        # fmt: off
+        points3d = np.array(
+            [
+                [1, 1, 0],
+                [-1, 1, 0],
+                [-2, 2, 0],
+                [-1, -1, 0],
+                [1, -1, 0],
+                [2, -2, 0],
+                [5, 5, 0]
+            ]
+        )
+        # fmt: on
+
+        for pt_3d in points3d:
+            sample_data.add_track(SfmTrack(pt_3d))
+
+        # Apply alignment transformation to sample_data
+        walignedTw = ellipsoid_utils.get_ortho_axis_alignment_transform(sample_data)
+        walignedTw = Similarity3(R=walignedTw.rotation(), t=walignedTw.translation(), s=1.0)
+        sample_data = sample_data.apply_Sim3(walignedTw)
+
+        # Verify correct 3d points.
+        computed_3d_points = np.array([sample_data.get_track(i).point3() for i in range(sample_data.number_tracks())])
+        expected_3d_points = np.array(
+            [
+                [0, -np.sqrt(2), 0],
+                [np.sqrt(2), 0, 0],
+                [2 * np.sqrt(2), 0, 0],
+                [0, np.sqrt(2), 0],
+                [-np.sqrt(2), 0, 0],
+                [-2 * np.sqrt(2), 0, 0],
+                [0, -5 * np.sqrt(2), 0],
+            ]
+        )
+        npt.assert_almost_equal(computed_3d_points, expected_3d_points, decimal=6)
+
+        # Verify correct camera poses.
+        expected_wTi_list = [
+            Pose3(walignedTw.rotation(), np.array([np.sqrt(2), 0, 2])),
+            Pose3(walignedTw.rotation(), np.array([0, -np.sqrt(2), 2])),
+            Pose3(walignedTw.rotation(), np.array([-np.sqrt(2), 0, 2])),
+        ]
+
+        computed_wTi_list = sample_data.get_camera_poses()
+        for wTi_computed, wTi_expected in zip(computed_wTi_list, expected_wTi_list):
+            assert wTi_computed.equals(wTi_expected, tol=1e-9)
+
+    def test_point_cloud_cameras_locked(self) -> None:
+        """Tests the get_ortho_axis_alignment_transform() function with a GtsfmData object containing 11 point cloud
+        points and 12 camera frustums from the door-12 dataset. Determines if the points are properly "locked" in
+        with one another before and after the alignment transformation is applied.
+        """
+
+        sample_data = GtsfmData(number_images=12)
+
+        # Read all 12 camera poses from data.mat file which contains extrinsics for all cameras in door12 dataset.
+        data = loadmat("data.mat")
+        M_list = [data["P"][0][i] for i in range(12)]
+        K = M_list[0][:3, :3]
+        Kinv = np.linalg.inv(K)
+        iTw_list = [Kinv @ M_list[i] for i in range(12)]
+        wTi_list = [Pose3(Rot3(iTw[:3, :3]), iTw[:, 3]).inverse() for iTw in iTw_list]
```

Comment:
Hi @adisingh50, thanks for the PR. It looks like some of this I/O code here is borrowed from the OlssonLoader. Can we just instantiate the OlssonLoader on the path to the Door test data (like we do in other unit tests), and use that object to get the poses you need?

Reply:
Yes, instantiating the OlssonLoader would certainly be cleaner and may fix the unit test error I was originally getting from directly reading from `data.mat`

Reply:
let's follow the example here: https://github.com/borglab/gtsfm/blob/master/tests/test_scene_optimizer.py#L27

Reply:
marking as resolved, thanks